pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -1308,7 +1308,7 @@ func (ctrl *PersistentVolumeController) provisionClaimOperation(claimObj interfa
 	tags[CloudVolumeCreatedForVolumeNameTag] = pvName
 
 	options := vol.VolumeOptions{
-		PersistentVolumeReclaimPolicy: v1.PersistentVolumeReclaimDelete,
+		PersistentVolumeReclaimPolicy: storageClass.ReclaimPolicy,
```

Comment:
tbh I'm not sure what happens in cluster upgrade and cases like that; an if statement here wouldn't hurt.